### ISSUE TITLE
Tags on module level not working

### ISSUE THREAD

**Issue by Tychodewaard** [dnnsoftware/Dnn.Platform#2883]
## Description of bug
The description of the tags on module settings give the impression that there is a dropdown with tags of if you start typing, tags from the vocabulary are populated. 
But...
1. this is not the case, you can just start typing
2. after typing any tag, there is no way to save this
3. what is populating are tags that are pagetags. So, tags that are put in the details of page settings. When you do this, another category is being generated in the vocabulary 'pagetags'

## Steps to reproduce
See steps above

## Current result
See steps above

## Expected result
The behaviour like the pagetags but also when tags are created in the other sections of the vocabulary. Alternatively, you could have a switch in the pagedetails-tags, where you can choose from what section the tags should be populated (default 'all')

## Affected version
* [x] 9.3.2
* [x] 9.3.1
But probably all 9 versions.

**Comment by sleupold** [comment]
The problem here is imcomplete implementation of taxonomy:
- vocabulary needs a selectType (predifined, folksonomy) where in "predefined" vocabularies, users may select from admin-defined items and in folksonomy might add own items as well
- whereever associating tags is offered, the admin should be able to specify the vocabulary to use. There might be specific vocabularies for dedicated areas of the site and some predefined vocabularies like "module types" should not be offered for tagging content.
- If the vocabulary is "predefined", a multiselect should be possible from existing list (type suggest) and same for "folksonomy", but if a non-existing item is associated, it should be added to the list.
In most cases, vocabularies to be used will be same for pages and modules and even some module items (if supported by the module).

**Comment by valadas** [comment]
I can reproduce in 9.4.0 RC

**Comment by MaiklT** [comment]
To give my two cents: Tags should be localizable...

**Comment by sleupold** [comment]
@MaiklT 
I agree in general, but it is not easy: who should maintain translations of tags - this might be possible for small taxonomies but would be a nightmare on folksonomies, unless using an automated translation service like Google or Bing translate. And what would be the benefit of having translated tags, if the content is not translated?

IMO a better approach would be language specific vocabularies, users may choose from.

**Comment by sleupold** [comment]
sounds still valid

**Comment by daguiler** [comment]
What should we do here?
Use the existing TAGS vocabulary?
Put module tags in a new MODULETAGS vocabulary?

**Comment by daguiler** [comment]
On a second look, I think it should behave just like the Tags input in the page settings, which lets you add new values which end up in the PAGETAGS vocabulary.

**Comment by Tychodewaard** [comment]
Tagging is one thing but next thing is that you want to actually use them :-) 
So, if you want a list of links to content with a certain tag, the list should be links to pages that have that tag and pages that have modules with that tag. Same for search results that are filtered with a tag.
Maybe there are other usecases but these are the ones that I can think of. I can imagine that for content that is served via the API, the scenario is different but I have ni experience with that. 

Does this help in any way where to put it? :-)
